src/createElement.ts

```diff
@@ -6,30 +6,53 @@ import {
   Component,
   MarkdownElement,
   MarkdownNode,
+  MarkdownPromise,
   MdFragmentType,
+  MdPromiseType,
 } from "./model";
 
+import { MdPromise } from ".";
+
 export function createElement(
   component: Component,
   attributes: Record<string, unknown> | null,
   ...children: MarkdownNode[]
 ): MarkdownNode | MarkdownNode[];
+export function createElement(
+  nodeType: typeof MdPromise,
+  attributes: null,
+  ...children: MarkdownPromise[]
+): MarkdownNode | MarkdownNode[];
 export function createElement(
   nodeType: string,
   attributes: null,
   ...children: MarkdownNode[]
 ): MarkdownNode | MarkdownNode[];
 export function createElement(
-  typeOrComponent: string | Component,
+  typeOrComponent: string | Component | typeof MdPromise,
   attributes: Record<string, unknown> | null,
-  ...children: MarkdownNode[]
+  ...children: MarkdownNode[] | MarkdownPromise[]
 ): MarkdownElement {
   if (typeOrComponent === MdFragmentType) {
-    return createFragmentElement(children);
+    return createFragmentElement(children as MarkdownNode[]);
+  }
+
+  if (typeOrComponent === MdPromise) {
+    // Throws error if there is more than one children.
+    if (children.flat().length > 1) {
+      throw new TypeError(
+        `<mdPromise> expected to receive a single promise that resolves to any markdown children.`
+      );
+    }
+    return createPromiseElement(children.flat()?.[0] as MarkdownPromise);
   }
 
   if (typeof typeOrComponent === "function") {
-    return createComponentElement(typeOrComponent, attributes, children);
+    return createComponentElement(
+      typeOrComponent as Component,
+      attributes,
+      children as MarkdownNode[]
```

Comment:
These assertions should not be necessary. What error do you get when leaving them out?